src/event_imp.rs

```diff
@@ -981,6 +986,450 @@ fn test_debug_ready() {
     assert_eq!("Writable", format!("{:?}", Ready::writable()));
 }
 
+#[cfg(any(
+    target_os = "dragonfly",
+    target_os = "freebsd",
+    target_os = "ios",
+    target_os = "macos"
+))]
+const AIO: usize = 0b01_0000;
+
+#[cfg(any(target_os = "freebsd"))]
+const LIO: usize = 0b10_0000;
+
+#[cfg(any(target_os = "linux", target_os = "android", target_os = "solaris"))]
+const PRI: usize = 0b100_0000;
+
+/// Interests used in registering.
+///
+/// Interests are used in registering [`Evented`] handles with [`Poll`],
+/// they indicate what readiness should be monitored for. For example if a
+/// socket is registered with readable interests and the socket becomes
+/// writable, no event will be returned from [`poll`].
+///
+/// [`Poll`]: struct.Poll.html                                                 
+/// [`readable`]: #method.readable
+/// [`writable`]: #method.writable
+#[derive(Copy, PartialEq, Eq, Clone, PartialOrd, Ord)]
+pub struct Interests(usize);
+
+impl Interests {
+    /// Returns `Interests` representing readable readiness.
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::readable();
+    ///
+    /// assert!(interest.is_readable());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    pub fn readable() -> Interests {
+        Interests(READABLE)
+    }
+
+    /// Returns `Interests` representing writable readiness.
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::writable();
+    ///
+    /// assert!(interest.is_writable());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    pub fn writable() -> Interests {
+        Interests(WRITABLE)
+    }
+
+    /// Returns `Interests` representing readable and writable readiness.
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::both();
+    ///
+    /// assert!(interest.is_readable());
+    /// assert!(interest.is_writable());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    pub fn both() -> Interests {
```

Comment:
This shouldn't be public because then `Interests` can extended in the future.

Reply:
Should `Interests::both()` even exist?
Maybe `Interests::all()` is better  since  `Interests` will be extended in the future.

Reply:
Lets make this private for now and we'll figure out `both` vs `all` in future pr.